src/Envs/classic_control/CartPole.jl

```diff
@@ -60,11 +60,11 @@ function CartPoleEnv()
 end
 
 function step!(env::CartPoleEnv, action)
-    # @assert action ∈ env.action_space "$action in ($(env.action_space)) invalid"
+    @assert action ∈ env.action_space "$action in ($(env.action_space)) invalid"
     state = env.state
     x, ẋ, θ, θ̇  = state[1:1], state[2:2], state[3:3], state[4:4]
-    # force = action == 2 ? env.force_mag : -env.force_mag
-    force = action .* env.force_mag  # action is +1 or -1
+    # Action is either 1 or 2. Force is either -force_mag or +force_mag
+    force = (2action-3) .* env.force_mag
```

Comment:
The DiffRL example gives the output as +/- 1. We'll have to transform that action as `(3-action)/2` before passing to `step!`.

Reply:
Ok, this turned out to be more challenging than expected😅.

Simply updating the `action()` function wasn't enough, as the program wasn't running and the error messages were not very helpful. A lot of troubleshooting later, I was finally able to isolate the problem, which was my assumption that the `action` is simply an Integer, and not a single element array. 

So fixed that, but there's a new error now, which is unavoidable. The `contains` function for the Discrete Space uses `try` and `catch` to find if the value being checked belongs to the Space. However, Zygote can not handle control flow jumps like `try` and `catch`. So the program crashes with a `Zygote.CompileError`. If the assertion check is removed, I get a very ugly Internal Error by buildbot, but the program runs, and takes ~200 episodes to train. 

How do we proceed?

Reply:
Pinging @tejank10

Reply:
We can use `@nograd` on the `contains` function, as its gradients are not required.